on-target-tests/tests/vreg_change_voltage.rs

```diff
@@ -0,0 +1,113 @@
+#![no_std]
+#![no_main]
+#![cfg(test)]
+
+use defmt_rtt as _; // defmt transport
+use defmt_test as _;
+use panic_probe as _; // panic handler
+use rp2040_hal as hal; // memory layout
+
+/// The linker will place this boot block at the start of our program image. We
+/// need this to help the ROM bootloader get our code up and running.
+/// Note: This boot block is not necessary when using a rp-hal based BSP
+/// as the BSPs already perform this step.
+#[link_section = ".boot2"]
+#[used]
+pub static BOOT2: [u8; 256] = rp2040_boot2::BOOT_LOADER_GENERIC_03H;
+
+/// External high-speed crystal on the Raspberry Pi Pico board is 12 MHz. Adjust
+/// if your board has a different frequency
+const XTAL_FREQ_HZ: u32 = 12_000_000u32;
+
+struct State {
+    vreg: hal::pac::VREG_AND_CHIP_RESET,
+}
+
+#[defmt_test::tests]
+mod tests {
+    use crate::{State, XTAL_FREQ_HZ};
+    use defmt::assert;
+    use defmt_rtt as _;
+    use panic_probe as _;
+    use rp2040_hal as hal;
+
+    use hal::pac::vreg_and_chip_reset::vreg::VSEL_A;
+    use hal::vreg;
+    use hal::{clocks::init_clocks_and_plls, pac, watchdog::Watchdog};
+
+    #[init]
+    fn setup() -> State {
+        unsafe {
+            hal::sio::spinlock_reset();
+        }
+        let mut pac = pac::Peripherals::take().unwrap();
+        let _core = pac::CorePeripherals::take().unwrap();
+        let mut watchdog = Watchdog::new(pac.WATCHDOG);
+
+        let _clocks = init_clocks_and_plls(
+            XTAL_FREQ_HZ,
+            pac.XOSC,
+            pac.CLOCKS,
+            pac.PLL_SYS,
+            pac.PLL_USB,
+            &mut pac.RESETS,
+            &mut watchdog,
+        )
+        .ok()
+        .unwrap();
+        State {
+            vreg: pac.VREG_AND_CHIP_RESET,
+        }
+    }
+
+    #[test]
+    fn change_onchip_regulator_voltage(state: &mut State) {
```

Comment:
How do we know that this test works reliably on every device?
The datasheet says: "Note that RP2040 may not operate reliably with its digital core supply (DVDD) at a voltage other than 1.1V."
Perhaps this test should only cover a smaller range of voltages, eg. 1.05 - 1.15V?

Reply:
In my experience, I have set the core voltage to 1.2V on more than 50 RP2040s and have never had a problem. Also, in the [thread about overclocking on the RP2040](https://forums.raspberrypi.com/viewtopic.php?t=301902), there is no example of a failure to increase the core voltage.  
However, this is not normal usage, there is no guarantee that it will work reliably on all RP2040s.

I noticed that with the default system clock, lowering the core voltage too much can cause problems. It should be a smaller test case range, as you mentioned. Especially in the direction of lowering the core voltage.  
1.20V should be fine, so i'll reduce it to 1.05 - 1.20V. 
If you are concerned, I would reduce it more. Reducing it is easy:)